### ISSUE TITLE
Implement a Language Server

### ISSUE THREAD

**Issue by mcchrish** [opral/inlang-sherlock#49]
Originally requested in i18n-ally but found out that it's not getting maintained these days.

https://github.com/lokalise/i18n-ally/issues/1146

But overall having a language server allow support to other editors that has LSP support.

**Comment by samuelstroschein** [comment]
Thoughts: 

* the language server makes sense in any case for linting messages, especially once we have ["Where used in code" plugin API](https://linear.app/opral/project/where-used-in-code-plugin-api-60f94d309ea1). 
* the value of LSP for third party sherlock implementations seems low. i wouldn't start on LSP for the sake of supporting sherlock for editor X.

**Comment by mcchrish** [comment]
Language server can also be a great conceptual choice to structure your code where "main" functionalities goes to the server and the vscode plugin will just be client making those functionalities available to vscode users. And you can of course  just continue maintaining the vscode plugin and can let other editor communities maintainer their own "plugin"/client.

This is the direction Tailwind have taken: https://github.com/tailwindlabs/tailwindcss-intellisense.
Which have given way for the neovim users to create their own plugin: https://github.com/luckasRanarison/tailwind-tools.nvim

**Comment by samuelstroschein** [comment]
yeah i understand the value behind LSP. ultimately, @felixhaeberle's decision. but lsp will likely not lead to third party sherlock extensions. the majority of code in sherlock doesn't concern itself with LSP related features
